src/wfs/src/main/java/org/geoserver/wfs/CatalogFeatureTypeCache.java

```diff
@@ -0,0 +1,56 @@
+/* Copyright (c) 2014 OpenPlans - www.openplans.org. All rights reserved.
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+*/
+package org.geoserver.wfs;
+
+import org.geoserver.catalog.Catalog;
+import org.geoserver.catalog.FeatureTypeInfo;
+import org.geotools.gml2.FeatureTypeCache;
+import org.geotools.util.logging.Logging;
+import org.opengis.feature.type.FeatureType;
+import org.opengis.feature.type.Name;
+
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Custom FeatureTypeCache that looks up directly from GeoServer catalog.
+ * <p>
+ * This cache class is used by XML bindings during parsing to obtain feature type information.
+ * This custom implementation allows for look up on demand, as opposed to pre-seeding the cache.
+ * </p>
+ */
```

Comment:
Ah so this was your anti-full scan patch.  Combining all our different changes into one commit was perhaps a bit confusing.